src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -163,9 +162,67 @@ mf::XWaylandWM::XWaylandWM(std::shared_ptr<WaylandConnector> wayland_connector,
         connection->_NET_ACTIVE_WINDOW,
         static_cast<xcb_window_t>(XCB_WINDOW_NONE));
 
+    connection->flush();
+
     cursors->apply_default_to(connection->root_window());
 
     wm_shell->surface_stack->add_observer(scene_observer);
+
+    // Detect and manage any windows that already exist
+    auto const query_tree_cookie = xcb_query_tree(*connection, connection->root_window());
+    auto const query_tree_reply = xcb_query_tree_reply(*connection, query_tree_cookie, nullptr);
```

Comment:
Can we use unique_ptr<> to ensure the free? We're calling out to C++ code that doesn't obviously guarantee noexecept.

Reply:
Doing it manually is consistent with the rest of the code. I agree some improvements are needed. Can we land this first and then I'll make another refactoring PR?

Reply:
Actually, big fancy solution seems to be more trouble than it's worth (at least at this time). Switched to using `unique_ptr` as you suggested.